From 08d2072d357cf2e32deb0a318d66a36e9322c19e Mon Sep 17 00:00:00 2001
From: Valentine Barshak <vbarshak@dev.rtsoft.ru>
Date: Wed, 31 Oct 2012 03:52:47 +0400
Subject: [PATCH 009/127] arm: tegra: ktt30: Add ADT7461 thermal sensor
 support

This adds ADT7461 I2C thermal sensor support.
Temperature alert interrupt is supported as well.

Signed-off-by: Valentine Barshak <vbarshak@dev.rtsoft.ru>
---
 arch/arm/mach-tegra/board-ktt30-sensors.c |   30 ++++++++++++++++++++++++++++-
 1 files changed, 29 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-tegra/board-ktt30-sensors.c b/arch/arm/mach-tegra/board-ktt30-sensors.c
index 648c076..d16cd52 100644
--- a/arch/arm/mach-tegra/board-ktt30-sensors.c
+++ b/arch/arm/mach-tegra/board-ktt30-sensors.c
@@ -33,6 +33,7 @@
 
 #include <linux/i2c.h>
 #include <linux/mpu.h>
+#include <linux/adt7461.h>
 #include <mach/fb.h>
 #include <mach/gpio.h>
 #include <generated/mach-types.h>
@@ -42,6 +43,31 @@
 #include "board-ktt30.h"
 
 
+static void ktt30_adt7461_alarm_fn(bool raised)
+{
+	if (raised)
+		pr_crit("WARNING! CRITICAL TEMPERATURE IS REACHED!\n");
+}
+
+static struct adt7461_platform_data ktt30_adt7461_pdata = {
+	.supported_hwrev = true,
+	.ext_range = false,
+	.therm2 = true,
+	.conv_rate = 0x05,
+	.offset = 0,
+	.hysteresis = 0,
+	.shutdown_ext_limit = 100,
+	.shutdown_local_limit = 100,
+	.throttling_ext_limit = 80,
+	.alarm_fn = ktt30_adt7461_alarm_fn,
+	.irq_gpio = TEGRA_GPIO_PCC2,
+};
+
+static struct i2c_board_info adt7461_board_info = {
+	I2C_BOARD_INFO("adt7461", 0x4C),
+	.irq = TEGRA_GPIO_TO_IRQ(TEGRA_GPIO_PCC2),
+	.platform_data = &ktt30_adt7461_pdata,
+};
 static struct mpu_platform_data mpu_gyro_data = {
 	.int_config	= 0x10,
 	.level_shifter	= 0,
@@ -117,5 +143,7 @@ static int mpuirq_init(void)
 
 int __init ktt30_sensors_init(void)
 {
-	return mpuirq_init();
+	i2c_register_board_info(4, &adt7461_board_info, 1);
+	mpuirq_init();
+	return 0;
 }
-- 
1.7.4.4

